src/lib/components/dialog/QDialog.svelte

```diff
@@ -1,77 +1,58 @@
 <script lang="ts">
   import { clickOutsideDialog } from "$lib/helpers";
-  import { createClasses } from "$lib/utils";
-  import { createEventDispatcher, onMount } from "svelte";
+  import { createEventDispatcher } from "svelte";
   import { QBtn } from "$lib";
   import type { QDialogProps } from "./props";
 
-  export let noBtn: QDialogProps["noBtn"] = false,
-    btnContent: QDialogProps["btnContent"] = undefined,
-    btnAttrs: QDialogProps["btnAttrs"] = {},
+  export let button: QDialogProps["button"] = false,
+    buttonLabel: QDialogProps["buttonLabel"] = undefined,
+    buttonProps: QDialogProps["buttonProps"] = {},
     position: QDialogProps["position"] = "default",
     modal: QDialogProps["modal"] = false,
     fullscreen: QDialogProps["fullscreen"] = false,
     persistent: QDialogProps["persistent"] = false,
-    userClasses: QDialogProps["userClasses"] = undefined;
+    userClasses: QDialogProps["userClasses"] = undefined,
+    value: QDialogProps["value"] = false;
   export { userClasses as class };
 
   const emit = createEventDispatcher();
-  let dialogElement: HTMLDialogElement;
+  let dialogElement: HTMLDialogElement | undefined;
 
-  let opened = false;
+  const DURATION_TOGGLE = 150;
 
-  onMount(() => {
-    opened = dialogElement.open;
-    dialogElement.style.display = opened ? "block" : "none";
-  });
-
-  $: canHideOnClickOutside = opened && persistent !== true;
-
-  $: positionClass = ["top", "right", "bottom", "left"].includes(position!) ? position : undefined;
-
-  $: classes = createClasses(
-    [opened && "active", positionClass, modal && "modal", fullscreen && "max"],
-    {
-      component: "q-dialog",
-      userClasses,
-    }
-  );
-
-  export function hide() {
-    if (dialogElement && dialogElement.open) {
+  $: if (dialogElement) {
+    if (value) {
+      dialogElement.style.display = "block";
+      modal ? dialogElement.showModal() : dialogElement.show();
+    } else {
       dialogElement.close();
-      opened = false;
       setTimeout(() => {
-        dialogElement.style.display = "none";
-      }, 250);
+        dialogElement!.style.display = "none";
+      }, DURATION_TOGGLE);
     }
   }
+
+  $: canHideOnClickOutside = value && !persistent;
+
+  export function hide() {
+    value = false;
```

Comment:
I think `value` should be set to false only if it's true in the first place to avoid running unnecessary svelte reactivity code.
Maybe, just in case, consider changing this to
```js
if(value) {
  value = false
}
```